### PR TITLE
Revoke statement document: Added Parameters section, converted Usage to Examples, updated SQL diagram

### LINKED ISSUE #492: sql: Clean up REVOKE
For the sake of consistency, add a **Parameters** section and convert **Usage** to **Examples**.

### PR DESCRIPTION
**Pull request by Amruta-Ranade**
Closes #492

### CODE REVIEW

**Comment by jseldess** [comment]
@Amruta-Ranade, sorry for the delay. I'm reviewing now.

File: generate/main.go [hunk 1]
```diff
@@ -423,7 +423,15 @@ func main() {
...
 				},
-				{name: "revoke_stmt", inline: []string{"privileges", "privilege_list", "privilege", "targets", "grantee_list"}},
+				{
+					name: "revoke_stmt", 
+					inline: []string{"privileges", "privilege_list", "privilege", "targets", "grantee_list"},
+					replace: map[string]string{
+						"table_pattern_list": "table_name ( ',' table_name )*",
+						"name_list": "database_name ( ',' database_name )*",
+						"'FROM' name ( ',' name )*": "'FROM' user_name ( ',' user_name )*"},
+					unlink: []string{"table_name","database_name"},
```
Comment (jseldess) [inline generate/main.go]: We should also unlink `"user_name"`.

**Comment by Amruta-Ranade** [comment]
@jseldess No problem! I fixed the issue. Please let me know if any changes are needed. Thanks!

**Comment by sploiselle** [comment]
:lgtm: mod one pedantic nit

---

Review status: 0 of 3 files reviewed at latest revision, 4 unresolved discussions, all commit checks successful.

---

*[revoke.md, line 25 at r1](https://reviewable.io:443/reviews/cockroachdb/docs/1595#-Kn0fqyYjPdG6Ro9m1rY:-Kn0fqyYjPdG6Ro9m1rZ:bngirnh) ([raw file](https://github.com/cockroachdb/docs/blob/e2b020d86196b14d08f3c8ec9079c11ff186e7a5/revoke.md#L25)):*
> ```Markdown
> Parameter | Description
> ----------|------------
> `table_name` | A comma-separated list of table names. Alternatively, to revoke privileges for all tables, use `*`. 
> ```

According to the diagram, `table_name` isn't a list (that's what the comma track implies). I'd make this more direct and simply say something like "The name of the table..."

---

*[revoke.md, line 26 at r1](https://reviewable.io:443/reviews/cockroachdb/docs/1595#-Kn0g8XePfa-Q5fOLsPW:-Kn0g8XePfa-Q5fOLsPX:b33azd1) ([raw file](https://github.com/cockroachdb/docs/blob/e2b020d86196b14d08f3c8ec9079c11ff186e7a5/revoke.md#L26)):*
> ```Markdown
> ----------|------------
> `table_name` | A comma-separated list of table names. Alternatively, to revoke privileges for all tables, use `*`. 
> `database_name` | A comma-separated list of database names.<br><br>Privileges revoked for databases will be revoked for any new tables created in the databases.
> ```

Same as `table_name` feedback above

---

*[revoke.md, line 27 at r1](https://reviewable.io:443/reviews/cockroachdb/docs/1595#-Kn0gCmsPd0gzQu47Oth:-Kn0gCmsPd0gzQu47Oti:b33azd1) ([raw file](https://github.com/cockroachdb/docs/blob/e2b020d86196b14d08f3c8ec9079c11ff186e7a5/revoke.md#L27)):*
> ```Markdown
> `table_name` | A comma-separated list of table names. Alternatively, to revoke privileges for all tables, use `*`. 
> `database_name` | A comma-separated list of database names.<br><br>Privileges revoked for databases will be revoked for any new tables created in the databases.
> `user_name` | A comma-separated list of [users](create-and-manage-users.html).
> ```

Same as `table_name` feedback above

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/docs/1595#-:-Kn0gLyZizBH11YVZk6x:b-mfisa6)*

**Comment by Amruta-Ranade** [comment]
@sploiselle Doesn't the comma-track imply a list? I checked documents for other statements (for example, the grant statement), and the comma-track seems to imply a list. Am I misunderstanding something?

**Comment by sploiselle** [comment]
@Amruta-Ranade This is probably confusing if we’re inconsistent in the documentation!

The idea behind the railroad diagrams is to show the “path” through the SQL syntax. If you follow that path and hit `table_name`, `database_name`, or `user_name`, what the SQL diagram expects is the name of an individual element (or `*` for all elements). To list multiple items, you follow the track through the optional comma track above it and re-enter an individual `table_name`, `database_name`, or `user_name`.

If you were to treat those elements as “comma-separated lists” themselves, under which circumstance would you ever divert from that element and proceed to the track with the optional comma? You never would, which means those values themselves represent individual names.

If you can point out where else you’ve seen this pattern in the docs, I’m glad to fix them so we’re consistent/accurate.

**Comment by Amruta-Ranade** [comment]
@sploiselle Hmm..it is indeed confusing :) Thank you for the clarification! I see the comma-track pattern in several documents: Drop Table, Drop View, Create Table, Show Grants. But I also see the pattern as you explain it in Select. And then a third pattern where we do mention in the Parameters section that the elements can be a comma-separated list, but not in the SQL diagram (example: Insert, Upsert). 

Please advise which pattern should I follow for the revoke statement. I think the second pattern is most suitable.

**Comment by jseldess** [comment]
@Amruta-Ranade, I think I'm guilty of the language you were emulating. Please follow @sploiselle's guidance, and we'll update the other docs in a separate PR.

**Comment by Amruta-Ranade** [comment]
@jseldess @sploiselle Actually, I can see the logic behind all three variations.  I agree the parameter definition can start with "The name of..." but I think we also need to state that optionally, a comma-separated list can be used. I hadn't understood the comma-track usage just from the SQL diagram till I read the parameter description, so a new user might be confused as well. Permission to use a fourth variation of the description please :)

**Comment by jseldess** [comment]
I think that sounds like a good, user-friendly idea, @Amruta-Ranade. @sploiselle, you good with that?

**Comment by sploiselle** [comment]
Sounds good.

**Comment by jseldess** [comment]
Looks good, @Amruta-Ranade. Thank you!

**Pull request merged**